platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/CopiedEncryptionService.java

```diff
@@ -0,0 +1,176 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.felix.cm.file;
+
+import java.io.File;
+
+import org.keyczar.Crypter;
+import org.keyczar.KeyczarTool;
+
+public class CopiedEncryptionService {
+    private final Crypter crypter;
+
+    private String getPasswordDirectory() {
+        final String karafHome = System.getProperty("karaf.home");
+        final String defaultPwDir =
+                "/Users/lambeaux/Connexta/instances/ddf-2.11.0-SNAPSHOT/etc/certs";
+
+        if (karafHome != null) {
+            File karafHomeDir = new File(karafHome);
+            if (!karafHomeDir.exists()) {
+                System.out.println("Karaf home DNE: " + karafHome);
+                return defaultPwDir;
+            }
+            if (!karafHomeDir.isDirectory()) {
+                System.out.println("Karaf home isn't directory: " + karafHome);
+                return defaultPwDir;
+            }
+            return karafHome.concat("/etc/certs");
+        }
+        System.out.println("Karaf home was null, defaulting to " + defaultPwDir);
+        return defaultPwDir;
+    }
+
+    public CopiedEncryptionService() {
+        System.out.println("KARAF HOME: " + System.getProperty("karaf.home"));
+        String passwordDirectory = getPasswordDirectory();
+
+        synchronized (CopiedEncryptionService.class) {
+            if (!new File(passwordDirectory.concat("/meta")).exists()) {
+                KeyczarTool.main(new String[] {"create", "--location=" + passwordDirectory,
+                        "--purpose=crypt", "--name=Password"});
+                KeyczarTool.main(new String[] {"addkey", "--location=" + passwordDirectory,
+                        "--status=primary"});
+            }
+            Crypter crypter = null;
+            try {
+                crypter = new Crypter(passwordDirectory);
+            } catch (Exception e) {
+                System.out.println(e.getMessage());
+            }
+
+            this.crypter = crypter;
+        }
+    }
+
+    /**
+     * Encrypts a plain text value using Keyczar.
+     *
+     * @param plainTextValue The value to encrypt.
+     */
+    public synchronized String encrypt(String plainTextValue) {
+        try {
+            return crypter.encrypt(plainTextValue);
+        } catch (Exception e) {
+            System.out.println("Key and encryption service failed to set up. Failed to encrypt. "
+                    + e.getMessage());
+            return plainTextValue;
+        }
+    }
+
+    /**
+     * Decrypts a plain text value using Keyczar
+     *
+     * @param encryptedValue The value to decrypt.
+     */
+    public synchronized String decrypt(String encryptedValue) {
+        try {
+            return crypter.decrypt(encryptedValue);
+        } catch (Exception e) {
+            System.out.println("Key and encryption service failed to set up. Failed to decrypt. "
```

Comment:
Why the `System.out.println`s instead of `LOGGER`?

Reply:
That'll fail checkstyle.

Reply:
I'll fix it. Wanted it to be very obvious that our stuff was getting used since I'm exploding felix and throwing in our own version of a class.